app/presenters/promo_rule_mailchimp_presenter.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module SpreeMailchimpEcommerce
+  module Presenters
+    class PromoRuleMailchimpPresenter
+      attr_reader :promotion
+
+      def initialize(promotion)
+        @promotion = promotion
+      end
+
+      # mailchimp require amount, type and target and its specific values,
+      # so for allow create promo_rule always, not only for proper promotion
+      # we send "dummy" values accepted by mailchimp
+      def json
+        {
+          id: Digest::MD5.hexdigest(promotion.id.to_s),
+          title: promotion.name || "",
+          description: promotion.description || "",
+          starts_at: starts_at,
+          ends_at: ends_at,
+          amount: amount.to_f,
+          type: type || "fixed",
+          target: target || "total",
+          created_at_foreign: promotion.created_at.strftime("%Y%m%dT%H%M%S"),
+          updated_at_foreign: promotion.updated_at.strftime("%Y%m%dT%H%M%S")
+        }.as_json
+      end
+
+      private
+
+      def starts_at
+        promotion.starts_at ? promotion.starts_at.strftime("%Y%m%dT%H%M%S") : ""
+      end
+
+      # with ends_at as "" the promo_code is unavaialbe but still exist
+      def ends_at
+        available? && promotion.expires_at ? promotion.expires_at.strftime("%Y%m%dT%H%M%S") : ""
+      end
+
+      def amount
+        if %w[total per_item].include? target
```

Comment:
if you use `if` without `else` it's better to do early return
`return unless %w[total per_item].include?(target)`

also, let's use parentheses to pass arguments here